### PR TITLE
Add a vault client

### PR DESCRIPTION
**Pull request by alvaroaleman**
This change adds a vaultclient because the upstream one has a couple of
shortcomings:
* It mostly has a "logical" api that always returns a
  map[string]interface{} instead of proper types and expects the user
  do know vault api paths
* It swallows 404s, which only makes sense for list requests
* A coupple of things can only be found out via multiple API calls, add
  simple helpers for those

Extracted from https://github.com/openshift/ci-tools/pull/1734
/cc @openshift/openshift-team-developer-productivity-test-platform 

ref https://issues.redhat.com/browse/DPTP-1891

### CODE REVIEW

File: pkg/vaultclient/vaultclient.go
Comment (alvaroaleman) [reply pkg/vaultclient/vaultclient.go]: fixed

File: pkg/vaultclient/vaultclient.go
Comment (alvaroaleman) [reply pkg/vaultclient/vaultclient.go]: fixed

File: pkg/vaultclient/vaultclient.go
Comment (alvaroaleman) [reply pkg/vaultclient/vaultclient.go]: They are both by themselves not sufficient, because `path` might not contain a slash. I've updated it similiar to your first suggestion and added some tests to capture the expectations

File: pkg/vaultclient/vaultclient.go
Comment (alvaroaleman) [reply pkg/vaultclient/vaultclient.go:102]: Yes, but we are talking about tiny amounts of data here, so I prefer readability over such optimizations. IIRC, json.Marshal will correctly handle a pointer to a nullpointer, so it might be possible to define `entity` as `*Entity` here. I can not easily test that without the other PR though, so I will leave it as-is for now and probably change in the other PR if that is indeed possible.

**Review by petr-muller — APPROVED** [review]
/hold

LGTM, holding for Bruno if necessary

**Pull request merged**
